internal/cli/project_destroy.go

```diff
@@ -0,0 +1,110 @@
+	clientpkg "github.com/hashicorp/waypoint/internal/client"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"strings"
+)
+
+type ProjectDestroyCommand struct {
+	*baseCommand
+
+	skipDestroyResources bool
+	confirm              bool
+}
+
+func (c *ProjectDestroyCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+	); err != nil {
+		return 1
+	}
+
+	err := c.DoApp(c.Ctx, func(ctx context.Context, app *clientpkg.App) error {
+		// Get the project we're destroying
+		project, err := c.project.Client().GetProject(ctx, &pb.GetProjectRequest{
+			Project: c.project.Ref(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Confirmation is required for destroying a project &/or its resources
+		if !c.confirm {
+			proceed, err := c.ui.Input(&terminal.Input{
+				Prompt: "Do you really want to destroy this project and its resources? Only 'yes' will be accepted to approve: ",
+				Style:  "",
+				Secret: false,
+			})
+			if err != nil {
+				c.ui.Output(
+					"Error getting input: %s",
+					clierrors.Humanize(err),
+					terminal.WithErrorStyle(),
+				)
+			} else if strings.ToLower(proceed) != "yes" {
+				app.UI.Output("Destroying project and resources requires confirmation.", terminal.WithWarningStyle())
+				return nil
+			}
+		}
+		err = app.DestroyProject(ctx, &pb.Job_DestroyProjectOp{
+			Project:              project.Project,
+			SkipDestroyResources: c.skipDestroyResources,
+		},
+		)
+		if err != nil {
+			return err
+		}
+		c.ui.Output("Project %q destroyed!", project.Project.Name, terminal.WithSuccessStyle())
+		return nil
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	return 0
+}
+
+func (c *ProjectDestroyCommand) Flags() *flag.Sets {
+	return c.flagSet(flagSetOperation, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "skip-destroy-resources",
+			Usage:   "Skips destroying resources created for the Waypoint project.",
+			Default: false,
+			Target:  &c.skipDestroyResources,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "auto-approve",
+			Usage:   "Auto-approve destroying the project & all resources. If unset, confirmation will be requested.",
+			Default: false,
+			Target:  &c.confirm,
+		})
+	})
+}
+
+func (c *ProjectDestroyCommand) Synopsis() string {
+	return "Delete the specified project."
+}
+
+func (c *ProjectDestroyCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint project destroy [options]
+
+  Delete the project and all resources created for all apps within the project.
+
+  You can optionally skip destroying the resources by setting
+  -skip-destroy-resources to true.
+
+  The workspace can continue to be used after this call, this just deletes
```

Comment:
I'm curious what it means to continue to use the workspace after? Is there another way to say this, I'm not sure what workspace we're referring to.

Reply:
I'm removing this for now, we want the operation to destroy all the resources for apps in a project in all workspaces where they're deployed.